src/scheduler/scheduler.rs

```diff
@@ -127,8 +127,11 @@ impl Scheduler {
     }
 
     pub fn main_loop(&self) {
-        self.block_manager.lock().and_then(|b| { b.sync(); Ok(()) });
         let mut org = self.peer_organizer.lock().unwrap();
+        let block_mgr = self.block_manager.lock().unwrap();
+        if let Some(task) = block_mgr.next_sync_task() {
+            org.schedule_to_free_peer(task);
+        }
```

Comment:
Yea scope of locks is something that we will revise in the future, for now it is okay to have it around the whole BlockManager. Btw having named all the things as Managers is not the proper way, I called it like that because in that point of time i needed placeholders and didn't care about naming, wdyt about renaming BlockManager into BlockchainSync?

Reply:
Right now the `BlockManager` is a typical manager object, in that it does several unrelated things (crafting requests, parsing requests, parsing responses, syncing). Instead of just renaming it, I'd rather refactor it into smaller objects. Maybe what is now `BlockManager` can become something like a facade class and also take care of the scope of locks?